epamticket/src/com/epam/ticket/dao/EpamTicketDAO.java

```diff
@@ -22,93 +25,95 @@
 
 public class EpamTicketDAO extends DefaultTicketDao {
 
+    public static final String SORT_REVERSE = "sortReverse";
+    public static final String SORT_NAME = "sortName";
     public static final Logger LOG = Logger.getLogger(EpamTicketDAO.class);
     public static final String QUERY_STRING = "SELECT {t:pk} FROM {CsTicket AS t} ";
-    private StringBuffer query;
-    private Set<EpamCsTicketFilter> availableFilters;
-    
-    public CsTicketModel getTicketById(String ticketId) {
-        List<CsTicketModel> csTicketModels = this.findTicketsById(ticketId);
-        if (csTicketModels.size() > 1) {
-            throw new AmbiguousIdentifierException("CsTicket with ticketId'" + ticketId + "' is not unique, " + csTicketModels.size() + " results!");
-        }
-        return csTicketModels.size() == 1 ? csTicketModels.get(0) : null;
-
-    }
+    protected Map<String, EpamCsSort> sorts = new HashMap<>();
 
-    public List<CsTicketModel> findTicketsByCriteria(EpamTicketSearchCriteria criteria) {
-        query = new StringBuffer(QUERY_STRING);
+    public List<CsTicketModel> findTicketsByCriteria(Map<String, String[]> searchCriteria, Set<EpamTicketsFilter> filters) {
+        StringBuilder query;
+        query = new StringBuilder(QUERY_STRING);
         Map<String, Object> paramMap = new TreeMap<>();
 
-        List<CsTicketPriority> priorities = criteria.getPriorities();
-        if (priorities != null && priorities.size() != 0) {
-            query.append(getJoiningString());
-            query.append("{priority} IN (?priority)");
-            paramMap.put("priority", priorities);
+        for (EpamTicketsFilter filter : filters) {
```

Comment:
I would prefer Java 8 way, but up to you
  filters.stream().forEach(filter -> {
            List<String> criterias = getCriterias(searchCriteria, filter.getName());
            if (!criterias.isEmpty()) {
                FilterSubqueryResult subQueryResult = filter.getFilterStrategy().buildFilterSubquery(filter, criterias);
                query.append(getFilterSubqueryOrEpmptyString(query.length(), subQueryResult));
                paramMap.putAll(subQueryResult.getQueryParams());
            }
        });